### PR TITLE
fix: respect CLAUDE_CONFIG_DIR for transcript directory

### PR DESCRIPTION
**Pull request by NodeJSmith**
### Respect `CLAUDE_CONFIG_DIR` for the transcript directory

- `DEFAULT_PROJECTS_DIR` was hardcoded to `~/.claude/projects`, so users running Claude Code with a custom `CLAUDE_CONFIG_DIR` had their transcripts silently missed by sync/import.
- `config.py` now resolves a `CLAUDE_CONFIG_DIR` constant (env var, falling back to `~/.claude`) and derives `DEFAULT_PROJECTS_DIR` from it, keeping the convention that `config.py` owns all path constants.
- `db.py` re-exports `DEFAULT_PROJECTS_DIR` from `config.py` for backward compatibility with downstream importers of that name.

## Summary by CodeRabbit

* **New Features**
  * Added support for customizing the Claude configuration directory through an environment setting.
  * Project transcript paths now automatically follow the configured Claude directory.

* **Bug Fixes**
  * Ensured database project paths remain consistent with the active configuration.

* **Tests**
  * Added coverage for custom and default configuration directory behavior.

### CODE REVIEW

File: src/ccrecall/config.py [hunk 1]
```diff
@@ -17,12 +17,18 @@
 
 from ccrecall.models import LOGGER_NAME
 
+# Claude Code config directory — respects CLAUDE_CONFIG_DIR when set.
+CLAUDE_CONFIG_DIR = Path(os.environ.get("CLAUDE_CONFIG_DIR", Path.home() / ".claude"))
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/config.py]: Fixed — using `or` instead of the default parameter so an empty `CLAUDE_CONFIG_DIR` falls back to `~/.claude`. Added a test for the empty-string edge case.

File: tests/test_db.py [hunk 2]
```diff
@@ -1440,3 +1441,55 @@ def test_memory_sync_does_not_import_heavy_deps(self):
...
+            f"assert DEFAULT_PROJECTS_DIR == Path({str(tmp_path)!r}) / 'projects', "
+            f"f'got {{DEFAULT_PROJECTS_DIR}}'\n"
+        )
+        result = subprocess.run(
+            [sys.executable, "-c", code],
+            capture_output=True,
+            text=True,
+            timeout=10,
+            env={**dict(os.environ), "CLAUDE_CONFIG_DIR": str(tmp_path)},
+        )
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/test_db.py:1463]: Fixed — added explicit `check=False` to all three `subprocess.run` calls.
